File: ObjC/AppmediationExample/AppmediationExample/Rewarded/RewardedView.m
//
//  RewardedView.m
//  AppmediationExample
//
//  Created by Mert Celik on 1/14/19.
//  Copyright © 2019 appmediation. All rights reserved.
//

#import "RewardedView.h"
#import <Appmediation/AMRewarded.h>

@interface RewardedView ()<AMRewardedDelegate>

@end

@implementation RewardedView

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.view setBackgroundColor:[UIColor whiteColor]];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    [self loadRewarded];
}

#pragma mark AMRewarded Implementation

- (void)loadRewarded {
    [AMRewarded setDelegate:self];
    [AMRewarded loadAd];
}

- (void)showRewarded {
    if ([AMRewarded isReady]) {
        [AMRewarded presentFromViewController:self];
    } else {
        NSLog(@"<AM>: Rewarded is not ready.");
    }
}

#pragma mark AMRewarded Delegates (Optional)

- (void)AMRewardedDidLoad:(AMRewarded *)rewarded {
    NSLog(@"<AM>: Rewarded did load.");
    [self.navigationController setNavigationBarHidden:YES animated:YES];
    [self showRewarded];
}

- (void)AMRewardedDidFailToLoad:(AMRewarded *)rewarded withError:(NSError *)error {
    NSLog(@"<AM>: Rewarded did fail to load with error: %@",error.localizedDescription);
}

- (void)AMRewardedDidShow:(AMRewarded *)rewarded {
    NSLog(@"<AM>: Rewarded did show.");
}

- (void)AMRewardedDidClick:(AMRewarded *)rewarded {
    NSLog(@"<AM>: Rewarded did click.");
}

- (void)AMRewardedDidClose:(AMRewarded *)rewarded {
    NSLog(@"<AM>: Rewarded did close.");
    [self.navigationController setNavigationBarHidden:NO animated:YES];
}

- (void)AMRewardedDidComplete:(AMRewarded *)rewarded withReward:(NSString *)name andAmount:(NSString *)amount {
    NSLog(@"<AM>: Rewarded did complete. User rewarded with %@ %@.",name,amount);
}

@end

File: Swift/AppmediationSwiftExample/AppmediationSwiftExample/Rewarded/RewardedView.swift
//
//  RewardedView.swift
//  AppmediationSwiftExample
//
//  Created by Mert Celik on 1/14/19.
//  Copyright © 2019 appmediation. All rights reserved.
//

import UIKit
import Appmediation.AMRewarded

class RewardedView: UIViewController, AMRewardedDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .white
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        loadRewarded()
    }
    
    // MARK: AMRewarded Implementation
    
    fileprivate func loadRewarded() {
        AMRewarded.setDelegate(self)
        AMRewarded.loadAd()
    }
    
    fileprivate func showRewarded() {
        if AMRewarded.isReady() {
            AMRewarded.present(from: self)
        } else {
            print("<AM>: Rewarded is not ready.")
        }
    }
    
    // MARK: AMRewarded Delegate (Optional)
    
    func amRewardedDidLoad(_ rewarded: AMRewarded!) {
        print("<AM>: Rewarded did load.")
        navigationController?.setNavigationBarHidden(true, animated: true)
        showRewarded()
    }
    
    func amRewardedDidFail(toLoad rewarded: AMRewarded!, withError error: Error!) {
        print("<AM>: Rewarded did fail to load with error: \(error.localizedDescription)")
    }
    
    func amRewardedDidShow(_ rewarded: AMRewarded!) {
        print("<AM>: Rewarded did show.")
    }
    
    func amRewardedDidClick(_ rewarded: AMRewarded!) {
        print("<AM>: Rewarded did click.")
    }
    
    func amRewardedDidClose(_ rewarded: AMRewarded!) {
        print("<AM>: Rewarded did close.")
        navigationController?.setNavigationBarHidden(false, animated: true)
    }
    
    func amRewardedDidComplete(_ rewarded: AMRewarded!, withReward name: String!, andAmount amount: String!) {
        print("<AM>: Rewarded did complete. User rewarded with " + name + amount + ".")
    }
    
}
